pkg/volume/cinder/cinder.go

```diff
@@ -35,11 +36,13 @@ import (
 
 // This is the primary entrypoint for volume plugins.
 func ProbeVolumePlugins() []volume.VolumePlugin {
-	return []volume.VolumePlugin{&cinderPlugin{nil}}
+	return []volume.VolumePlugin{&cinderPlugin{}}
 }
 
 type cinderPlugin struct {
 	host volume.VolumeHost
+	// Guarding SetUp and TearDown operations
+	volumeGuard keymutex.KeyMutex
```

Comment:
Also, I'm looking at KeyMutex and this is the exact thing we need to stripe locks by pv/pvc name.